idea/src/org/jetbrains/kotlin/idea/intentions/ConvertTwoComparisonsToRangeCheckIntention.kt

```diff
@@ -0,0 +1,124 @@
+                            generateRangeExpressionData(firstLeft, secondRight, firstRight, minDecrementByOne = firstOpToken == KtTokens.LT, maxDecrementByOne = secondOpToken == KtTokens.GT)
+                        firstRight !is KtConstantExpression && firstRight.evaluatesTo(secondRight) ->
+                            generateRangeExpressionData(firstRight, firstLeft, secondLeft, minDecrementByOne = firstOpToken == KtTokens.LT, maxDecrementByOne = secondOpToken == KtTokens.GT)
+                        else -> null
+                    }
+                    KtTokens.LTEQ, KtTokens.LT -> when {
+                        firstLeft !is KtConstantExpression && firstLeft.evaluatesTo(secondRight) ->
+                            generateRangeExpressionData(firstLeft, secondLeft, firstRight, minDecrementByOne = firstOpToken == KtTokens.LT, maxDecrementByOne = secondOpToken == KtTokens.LT)
+                        firstRight !is KtConstantExpression && firstRight.evaluatesTo(secondLeft) ->
+                            generateRangeExpressionData(firstRight, firstLeft, secondRight, minDecrementByOne = firstOpToken == KtTokens.LT, maxDecrementByOne = secondOpToken == KtTokens.LT)
+                        else -> null
+                    }
+                    else -> null
+                }
+            }
+            else -> null
+        }
+    }
+
+    private fun generateRangeExpressionData(value: KtExpression, min: KtExpression, max: KtExpression, minDecrementByOne: Boolean = false, maxDecrementByOne: Boolean = false): RangeExpressionData? {
+        if (minDecrementByOne || maxDecrementByOne) {
+            val type = value.getType(value.analyze()) ?: return null
+            if (!KotlinBuiltIns.isInt(type) && !KotlinBuiltIns.isLong(type) && !KotlinBuiltIns.isShort(type) && !KotlinBuiltIns.isChar(type)) return null
+        }
+
+        val minText = if (minDecrementByOne) min.getDecrementByOneString() else min.text
+        val maxText = if (maxDecrementByOne) max.getDecrementByOneString() else max.text
+        return RangeExpressionData(value.text, minText ?: return null, maxText ?: return null)
+    }
+
+    private fun KtExpression.getDecrementByOneString(): String? {
+        val type = getType(analyze()) ?: return null
+
+        when (this) {
+            is KtConstantExpression -> {
+                val number: Number = when {
+                    KotlinBuiltIns.isInt(type) -> text.toInt() - 1
+                    KotlinBuiltIns.isLong(type) -> {
+                        val text = text
+                        val longText = if (text.endsWith("l") || text.endsWith("L")) text.substring(0, text.length - 1) else text
+                        longText.toLong() - 1
+                    }
+                    KotlinBuiltIns.isShort(type) -> java.lang.Short.parseShort(text) - 1
+                    KotlinBuiltIns.isChar(type) -> text[0].toInt() - 1
```

Comment:
I think it is better to replace character with character, either explicit e.g. `'z' - 1` or implicit `'y'`